delphi/app/routes.py

```diff
@@ -4,14 +4,15 @@
 
 @app.route('/update-sensor-data', methods=['POST'])
 def updateSensorData():
-    data = request.get_json(silent=True)
+    try:
+        data = request.get_json()
+        sensorId = int(data.get('id'))
+        count = int(data.get('count'))
 
-    if isinstance(data, dict):
-        sensorId = data.get('id')
-        count = data.get('count')
-
-        if isinstance(sensorId, int) and isinstance(count, int):
+        if sensorId is not None and count is not None:
```

Comment:
Checking for `not None` here might not be useful. When casting `sensorId` or `count` to an int, I don't believe `None` is a state they can be in. If the cast fails, it will throw an exception

Reply:
You're right in that it would throw an exception. However, requests wouldn't be accepted with a count of 0 if you're saying I should opt for `if sensorId and count:`

The tests have been updated to verify this behavior.

Reply:
I'm saying you don't need to do the check

Reply:
![image](https://user-images.githubusercontent.com/31143072/75737537-9c5ee980-5cc5-11ea-86f1-3aa7d81a0486.png)